boa/src/builtins/function/mod.rs

```diff
@@ -513,15 +513,26 @@ pub fn make_constructor_fn(
 ///     some other number of arguments.
 ///
 /// If no length is provided, the length will be set to 0.
-pub fn make_builtin_fn<N>(function: NativeFunctionData, name: N, parent: &Value, length: usize)
-where
+pub fn make_builtin_fn<N>(
+    function: NativeFunctionData,
+    name: N,
+    parent: &Value,
+    length: usize,
+    interpreter: &Interpreter,
+) where
     N: Into<String>,
 {
     let name = name.into();
     let _timer = BoaProfiler::global().start_event(&format!("make_builtin_fn: {}", &name), "init");
 
-    // FIXME: function needs the Function prototype set.
-    let mut function = Object::function(Function::builtin(Vec::new(), function), Value::null());
+    let mut function = Object::function(
+        Function::builtin(Vec::new(), function),
+        interpreter
+            .global()
+            .get_field("Function")
+            .get_field("prototype"),
```

Comment:
I'm not super up to date with this, but isn't this similar to the `prototype()` function?

https://github.com/boa-dev/boa/blob/8fd5533460af8c0c424f6ddf4f3d45ed84815f8f/boa/src/builtins/object/mod.rs#L432

Reply:
Yes, it looks like the final `get_field` does the same thing as the `prototype()` function.
Thanks for your comment.

Reply:
> I'm not super up to date with this, but isn't this similar to the `prototype()` function?

Its not the same. the `.prototype()` function does not return the `prototype` it returns the prototype instance so for `Function` it would be `Object.prototype`, not `Function.protoype`

It is confusing, it should be called `prototype_instance()` or something like that, since that is what is does

Reply:
Created an issue about the naming of `prototype` #607

Reply:
> for `Function` object it would be `Object.prototype`, not `Function.protoype`

Shouldn't `Function.__proto__ === Function.prototype`?

Reply:
> Shouldn't `Function.__proto__ === Function.prototype`?

Actually you are right, sorry about that :sweat_smile: , the `Function` object has the `__proto__` set to `Function.prototype` since it is a function and it inherits from it self (`Function`). so in this case it would be the same.
![Screenshot_2020-08-07_18-48-30](https://user-images.githubusercontent.com/8566042/89668776-a2916b80-d8de-11ea-9066-e4bdfff0a6ed.png)

Reply:
This turns out to be a 'special case'. :)

Reply:
Since @HalidOdat mentioned about updating parts of the code to cache frequently used objects such as `Function`, `Object`, etc. can we leave it this way for now?

Reply:
> Since @HalidOdat mentioned about updating parts of the code to cache frequently used objects such as `Function`, `Object`, etc. can we leave it this way for now?

Yes. I was just suggesting a possible optimization that we can do in the future.